subprojects/packagefiles/sdl2/meson.build

```diff
@@ -367,11 +368,28 @@ endif
 core_deps = [
     cc.find_library('m', required : false),
     cc.find_library('dl', required : false),
-    cc.find_library('iconv', required : false),
     dbus_dep,
     ibus_dep,
 ]
 
+if not opt_system_iconv.disabled()
+    if meson.version().version_compare('>=0.60')
+        iconv_dep = dependency('iconv', required : opt_system_iconv)
+    else
+        iconv_dep = cc.find_library('iconv', required : opt_system_iconv)
```

Comment:
If iconv is required, but it's available as a built-in libc thing, this will error out because find_library() fails.